pkg/serverless/metrics/enhanced_metrics.go

```diff
@@ -48,37 +50,37 @@ func getOutOfMemorySubstrings() []string {
 }
 
 // GenerateRuntimeDurationMetric generates the runtime duration metric
-func GenerateRuntimeDurationMetric(start time.Time, end time.Time, status string, tags []string, metricsChan chan []metrics.MetricSample) {
+func GenerateRuntimeDurationMetric(start time.Time, end time.Time, status string, tags []string, demux aggregator.Demultiplexer) {
 	// first check if both date are set
 	if start.IsZero() || end.IsZero() {
 		log.Debug("Impossible to compute aws.lambda.enhanced.runtime_duration due to an invalid interval")
 	} else {
 		duration := end.Sub(start).Milliseconds()
-		metricsChan <- []metrics.MetricSample{{
+		demux.AddTimeSamples(0, []metrics.MetricSample{{
 			Name:       runtimeDurationMetric,
 			Value:      float64(duration),
 			Mtype:      metrics.DistributionType,
 			Tags:       tags,
 			SampleRate: 1,
-			Timestamp:  float64(end.UnixNano()),
-		}}
+			Timestamp:  float64(end.UnixNano()) / float64(time.Second),
```

Comment:
Was there a bug before your change here?

Reply:
No bug here before, the Serverless Agent was sending it nano and / second later in the process, I've normalized to seconds everywhere.